mantis-common/src/main/java/io/mantisrx/runtime/descriptor/StageSchedulingInfo.java

```diff
@@ -150,4 +159,69 @@ public boolean equals(Object obj) {
             return false;
         return true;
     }
+
+    public static class Builder {
+        private int numberOfInstances;
+        private MachineDefinition machineDefinition;
+        private List<JobConstraints> hardConstraints;
+        private List<JobConstraints> softConstraints;
+        private StageScalingPolicy scalingPolicy;
+        private boolean scalable;
+        private boolean inheritInstanceCount;
+
+        public Builder setNumberOfInstances(int numberOfInstances) {
+            this.numberOfInstances = numberOfInstances;
+            return this;
+        }
+
+        public Builder setOptionalMachineDefinition(MachineDefinition machineDefinition) {
+            this.machineDefinition = machineDefinition;
+            return this;
+        }
+
+        public Builder setOptionalHardConstraints(List<JobConstraints> hardConstraints) {
+            this.hardConstraints = hardConstraints;
+            return this;
+        }
+
+        public Builder setOptionalSoftConstraints(List<JobConstraints> softConstraints) {
+            this.softConstraints = softConstraints;
+            return this;
+        }
+
+        public Builder setOptionalScalingPolicy(StageScalingPolicy scalingPolicy) {
```

Comment:
This feels unrelated to this diff. I see a lot of tests have changed as a result of this change. In general, I think it's a good idea to keep refactoring changes isolated from business logic changes. I would put this up as a separate diff. We can also explore using Lombok annotations for simplifying a lot of this code in that diff.

Reply:
I used the auto builder generator from IntelliJ thus it auto-converted all the usages.
Definitely +1 on lambok.